db/arena_wrapped_db_iter.cc

```diff
@@ -64,8 +64,54 @@ void ArenaWrappedDBIter::Init(
   memtable_range_tombstone_iter_ = nullptr;
 }
 
+void ArenaWrappedDBIter::MaybeAutoRefresh() {
+  if (cfh_ != nullptr && allow_refresh_ &&
+      read_options_.auto_refresh_iterator_enabled) {
+    uint64_t cur_sv_number = cfh_->cfd()->GetSuperVersionNumberRelaxed();
+    if (sv_number_ != cur_sv_number) {
+      Slice key = nullptr;
+      if (db_iter_->Valid()) {
+        key = db_iter_->key();
+      }
+
+      DoRefresh(read_options_.snapshot, cur_sv_number);
+
+      if (key != nullptr) {
+        db_iter_->Seek(key);
```

Comment:
The key() Slice is valid until the call to the next iterator modifying operation. It's not clear to me that the previous key() Slice will remain valid through DoRefresh() and much of Seek(). Since this is a rare, heavy-weight operation anyway, we should copy the key into a temporary string to avoid such issues.